docs/src/main/asciidoc/kafka-streams.adoc

```diff
@@ -1113,6 +1113,55 @@ CMD ["./application", "-Dquarkus.http.host=0.0.0.0", "-Xmx32m"]
 Now start Docker Compose as described above
 (don't forget to rebuild the container images).
 
+== Kafka Streams Health Check
+
+If you are using the `quarkus-smallrye-health` extension, `quarkus-kafka-streams` will automatically add:
+
+* a readiness health check to validate that all topics declared in the `quarkus.kafka-streams.topics` property are created,
+* a liveness health check based on the Kafka Streams state.
+
+So when you access the `/health` endpoint of your application you will have information about the state of the KStreams and the available and/or missing topics.
+
+This is an example of `DOWN` status:
+[source, subs=attributes+]
+----
+http http://aggregator:8080/health
+
+HTTP/1.1 503 Service Unavailable
+content-length: 490
+content-type: application/json; charset=UTF-8
+
+{
+    "status": "DOWN",
```

Comment:
So I'm still a bit skeptical about mixing these two thins in one health check. I will effectively make it impossible to set up a K8s liveness probe for restarting the service after a stream processing error, as also non-existing topics may trigger the restart. Would it be possible to set up two distinct health checks perhaps?

Reply:
hmmm... I'm affraid I don't understand what you mean.
If you base your liveness probe on `/health/live` then K8S will only restart the pod if streams state is down (a stream processing error for instance) not when topics are non-existing.

Reply:
Ah, ok. Sorry, that wasn't apparent to me, if the two checks can be queried independently, then all is good.

Reply:
Maybe the adoc is not clear enough ?

Reply:
Yes, it might be a good idea to show how the two can be queried independently.